ui/src/components/helpers/TipTap/FillInBlank.js

```diff
@@ -0,0 +1,144 @@
+import {
+  Node,
+  mergeAttributes,
+  nodeInputRule,
+  nodePasteRule,
+} from '@tiptap/core';
+import { NodeViewWrapper, ReactNodeViewRenderer } from '@tiptap/react';
+import React, { useState, useEffect, useRef } from 'react';
+
+export const FillInBlankNode = Node.create({
+  name: 'fillInBlank',
+  content: 'text*',
+  group: 'inline',
+  inline: true,
+  atom: true,
+
+  addAttributes() {
+    return {
+      editorText: {
+        default: null,
+      },
+      userText: {
+        default: null,
+      },
+    };
+  },
+  addCommands() {
+    return {
+      insertFillInBlank:
+        () =>
+        ({ commands, editor }) => {
+          const position = editor.state.selection.from;
+          return commands.insertContentAt(position, {
+            type: 'fillInBlank',
+          });
+        },
+      filledInBlank:
+        (editorText) =>
+        ({ commands }) => {
+          return commands.setNodeAttribute('editorText', editorText);
+        },
+    };
+  },
+  addInputRules() {
+    return [
+      nodeInputRule({
+        find: /(\_{4}\(([^)]+)\))/,
+        type: this.type,
+        getAttributes: (match) => {
+          return {
+            editorText: match[2],
+          };
+        },
+      }),
+    ];
+  },
+  addPasteRules() {
+    return [
+      nodePasteRule({
+        find: /(\_{4}\(([^)]+)\))/,
+        type: this.type,
+        getAttributes: (match) => {
+          return {
+            editorText: match[2],
+          };
+        },
+      }),
+    ];
+  },
+
+  toDOM() {
+    return ['span', { class: 'fillInBlank' }, 0];
+  },
+
+  parseHTML() {
+    return [
+      {
+        tag: 'span',
+      },
+    ];
+  },
+
+  renderHTML({ HTMLAttributes }) {
+    return [
+      'span',
+      { class: 'fillInBlank', ...HTMLAttributes },
+      0,
+      // Remove the trailing break element
+    ];
+  },
+  addNodeView() {
+    return ReactNodeViewRenderer((props) => {
+      const inputRef = useRef(null);
+      const [userText, setUserText] = useState(props.node.attrs.userText);
+
+      useEffect(() => {
+        if (inputRef.current) {
+          let textWidth = getTextWidth(userText);
+          if (!userText) {
+            textWidth = getTextWidth('Blank');
+          }
+          inputRef.current.style.width = `${textWidth}px`;
+        }
+      }, [userText]);
```

Comment:
Check dependency array if it throw warning

Reply:
It doesn't throw any warning